relay-server/src/actors/project_cache.rs

```diff
@@ -899,20 +983,29 @@ impl Service for ProjectCacheService {
                 buffer_guard,
                 index: BTreeMap::new(),
                 buffer,
+                global_config,
             };
 
             loop {
                 tokio::select! {
-                    biased;
-
-                    Some(message) = state_rx.recv() => broker.merge_state(message),
-                    // Buffer will not dequeue the envelopes from the spool if there is not enough
-                    // permits in `BufferGuard` available. Currently this is 50%.
-                    Some(managed_envelope) = buffer_rx.recv() => broker.handle_processing(managed_envelope),
-                    _ = ticker.tick() => broker.evict_stale_project_caches(),
-                    Some(message) = rx.recv() => broker.handle_message(message),
-                    else => break,
+                                biased;
+
+                                    Ok(()) = subscription.changed() => {
```

Comment:
Why the formatting looks a bit off? 

And also wondering if we still want to be biased for processing the incoming messages first.

Reply:
i thought my rustformatter was broken so i'd wait a bit later until it does, but maybe it's not supposed to work within a macro ? I'll fix it manually now